target-app/Home/Views/HomeViewController.swift

```diff
@@ -137,7 +160,37 @@ extension HomeViewController: BottomSheetPresenter {
     }
 }
 
-extension HomeViewController {
+extension HomeViewController: MKMapViewDelegate{
+    
+    func mapView(_ mapView: MKMapView, viewFor annotation: MKAnnotation) -> MKAnnotationView? {
+        guard annotation is MKPointAnnotation else { print("no mkpointannotaions"); return nil }
+        
+        let reuseId = "pin"
+        var pinView = mapView.dequeueReusableAnnotationView(withIdentifier: reuseId) as? MKPinAnnotationView
+        
+        if pinView == nil {
+            pinView = MKPinAnnotationView(annotation: annotation, reuseIdentifier: reuseId)
```

Comment:
instead you could do
```suggestion
        let pinView = mapView.dequeueReusableAnnotationView(withIdentifier: reuseId) as? MKPinAnnotationView 
          ??  MKPinAnnotationView(annotation: annotation, reuseIdentifier: reuseId)
```

Reply:
Same comment as above. If I do this, it seems that `pinView` variable is no longer accessible from `else` statement and `return`.